### ISSUE TITLE
WCAG 2.1.2 - No Keyboard Trap

### ISSUE THREAD

**Issue by tbostic32** [mitre/demodocus#18]
S.C. 2.1.2 - If keyboard focus can be moved to a component of the [content] using a keyboard interface, then focus can be moved away from that component using only a keyboard interface. 

Test Procedure - We would expect our tab order to be a "loop" where the last element reached is also the first element. If we find other cycles, we should flag those as possible keyboard traps.

Reporting - If an unexpected cycle is found within the tab order, we will mark the element at the start of that cycle as the offending element.

Use the standard data format. Append a violation onto the states violations. Specifically, for now we want:

* `type = "atomic`
* `level = "warning"`
* `category = "S.C. 2.1.2"`
* `element = {element info}`
